File: src/main/com/jason/s7/KtBase133.java
package com.jason.s7;

public class KtBase133 {
    public static void main(String[] args) {
        /**
         * Java端調用Kotlin方法，但Java無法使用Kotlin方法的默認參數值
         */
//        KtBase133Kt.showInfo("Jason");

        /**
         * 相當於Java使用Kotlin的默認參數
         */
        KtBase133Kt.showInfo2("Juan");
    }
}

File: src/main/com/jason/s7/KtBase133.kt
package com.jason.s7

//註解@JvmOverloads與Kotlin
fun main() {
    showInfo("Jason")
    showInfo2("Juan")
}

fun showInfo(name: String, age: Int = 20, sex: Char = 'M') = println("name=$name, age=$age, sex=$sex")

/**
 * 使用@JvmOverloads，編譯階段會專門重載一個函數給Java使用
 */
@JvmOverloads
fun showInfo2(name: String, sex: Char = 'F') = println("name=$name, sex=$sex")

